src/main/java/frc/robot/subsystems/Maflipulator.java

```diff
@@ -0,0 +1,193 @@
+        if (currentPosition == MaflipulatorSide.kFront) {
+            if (flipSpeed > 0 && anglePot.get() >= FRONT_UPPER_ANGLE) {
+                flipSpeed = 0;
+            }
+            if (flipSpeed < 0 && anglePot.get() <= FRONT_LOWER_ANGLE) {
+                flipSpeed = 0;
+            }
+        } else {
+            if (flipSpeed > 0 && anglePot.get() <= BACK_UPPER_ANGLE) {
+                flipSpeed = 0;
+            }
+            if (flipSpeed < 0 && anglePot.get() >= BACK_LOWER_ANGLE) {
+                flipSpeed = 0;
+            }
+        }
+        return flipSpeed;
+    }
+
+    public Command restrictRotate() {
+        // The command is named "Restrict Rotate" and requires this subsystem.
+        return new Command("Restrict Rotate", this) {
+
+            @Override
+            protected void execute() {
+                double flipSpeed = Robot.xBoxCoPilot.getY(Hand.kRight);
+                flipSpeed = restrict(flipSpeed);
+                flipMotor.set(flipSpeed);
+            }
+
+            @Override
+            protected boolean isFinished() {
+                return false;
+            }
+
+        };
+    }
+
+    public Command Flip() {
+        return new InstantCommand("Flip", this, () -> {
+
+            Command moveCommand;
+            if (currentPosition == MaflipulatorSide.kFront) {
+                moveCommand = moveToPosition(270);
+                currentPosition = MaflipulatorSide.kBack;
+            } else {
+                moveCommand = moveToPosition(90);
+                currentPosition = MaflipulatorSide.kFront;
+            }
+
+            Scheduler.getInstance().add(moveCommand);
+        });
+    }
+
+    public Command PIDScoringPosition() {
+        return new InstantCommand("PID Scoring Position", this, () -> {
+
+            Command moveCommand;
+            if (currentPosition == MaflipulatorSide.kFront)
+                moveCommand = moveToPosition(90);
+            else
+                moveCommand = moveToPosition(270);
+
+            Scheduler.getInstance().add(moveCommand);
+        });
+
+    }
+
+    public Command PIDPickupPosition() {
+        return new InstantCommand("PID Pickup Position", this, () -> {
+
+            Command moveCommand;
+            if (currentPosition == MaflipulatorSide.kFront)
+                moveCommand = moveToPosition(FRONT_LOWER_ANGLE);
+            else
+                moveCommand = moveToPosition(BACK_LOWER_ANGLE);
+
+            Scheduler.getInstance().add(moveCommand);
+        });
+    }
+
+    public Command moveToPosition(double targetPosition) {
+        return new PIDCommand("Move to Position", .01, .0009, 0.0, this) {
+
+            @Override
+            protected void initialize() {
+                anglePID.reset();
+                anglePID.setSetpoint(targetPosition);
+                anglePID.enable();
+            }
+
+            @Override
+            protected boolean isFinished() {
+                return anglePID.onTarget();
```

Comment:
Look at adding a `&& Math.abs(angleCorrection) < 0.2` or similar, to make sure it stops at the right location